core/src/io/anuke/mindustry/world/blocks/power/PowerNode.java

```diff
@@ -299,4 +317,27 @@ protected void drawLaser(Tile tile, Tile target){
         Draw.color();
     }
 
+    public static boolean insulated(Tile tile, Tile other){
+        return insulators(tile, other).size > 0;
+    }
+
+    public static Array<Tile> insulators(Tile tile, Tile other) {
+        Array<Tile> tiles = new Array<Tile>();
+        Array<Point2>points = PlaceUtils.normalizeDiagonal(tile.x, tile.y, other.x, other.y);
+
+        for(int i = 0; i < points.size; i++){
+            Point2 point = points.get(i);
+            Block block = world.tile(point.x, point.y).block();
+
+            if (block instanceof BlockPart){
```

Comment:
Don't check for blockparts manually. Use `world.ltile(point.x, point.y)` to return the linked tile or null.

Reply:
Changed as per your code example above 😎